src/v/cloud_topics/level_one/domain_supervisor.cc

```diff
@@ -0,0 +1,198 @@
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "cloud_topics/level_one/domain_supervisor.h"
+
+#include "cloud_topics/logger.h"
+#include "cluster/controller.h"
+#include "cluster/topics_frontend.h"
+#include "cluster/types.h"
+#include "model/fundamental.h"
+#include "model/namespace.h"
+
+#include <seastar/coroutine/as_future.hh>
+
+using namespace std::chrono_literals;
+
+namespace experimental::cloud_topics::l1 {
+
+class domain_supervisor::impl {
+    // Intentionally under the ss::sstring SSO size.
+    static constexpr model::topic_view domain_topic_name = model::topic_view{
+      std::string_view{"ct_l1_domain"}};
+
+public:
+    explicit impl(cluster::controller* controller)
+      : _controller(controller) {}
+
+    ss::future<> start() {
+        if (ss::this_shard_id() == 0) {
+            _as = {};
+            _loop = do_topic_reconciliation_loop();
+        }
+        // TODO(cloud-topics): We should also create domain supervisors if this
+        // shard owns a domain partition.
+        co_return;
+    }
+
+    ss::future<> stop() {
+        if (ss::this_shard_id() == 0 && _loop) {
+            _as.request_abort();
+            co_await *std::exchange(_loop, std::nullopt);
+        }
+    }
+
+private:
+    ss::future<> do_topic_reconciliation_loop() {
+        while (!_as.abort_requested()) {
+            bool aborted = co_await loop_sleep();
+            if (aborted) {
+                // If we were aborted, we exit the loop.
+                co_return;
+            }
+            if (_controller->get_topics_state().local().contains(
+                  model::topic_namespace{
+                    model::kafka_internal_namespace, domain_topic_name})) {
+                co_await ensure_domains_replication_factor();
+            } else {
+                co_await create_domains_topic();
+            }
+        }
+    }
+
+    ss::future<bool> loop_sleep() {
+        constexpr auto interval = 10min;
+        simple_time_jitter<ss::lowres_clock> jitter(interval);
+        try {
+            co_await ss::sleep_abortable<ss::lowres_clock>(
+              jitter.next_duration(), _as);
+            co_return true;
+        } catch (const ss::sleep_aborted& ex) {
+            // do nothing, the caller will handle exiting properly.
+            std::ignore = ex;
+            co_return false;
+        }
+    }
+
+    ss::future<> ensure_domains_replication_factor() {
+        auto tp_ns = model::topic_namespace{
+          model::kafka_internal_namespace, domain_topic_name};
+        auto rf = _controller->get_topics_state()
+                    .local()
+                    .get_topic_replication_factor(tp_ns);
+        if (!rf) {
+            vlog(cd_log.warn, "unable to find {} replication factor", tp_ns);
+            co_return;
+        }
+        auto target_rf = cluster::replication_factor(
+          _controller->internal_topic_replication());
```

Comment:
Do I understand correctly that this will ensure that the RF of the domain toic is the same as the RF of the controller? What was the motivation for this?

The RF will be updated every 10m, if we can afford to wait 10m we can probably afford not having 1:1 mapping between number of brokers and replication factor.

Reply:
No this is the cluster config for rf of topics in kafka_internal namespace.

Reply:
oh, if this is the case why do we need the background fiber at all? it is possible to subscribe to the config property 
something like this
```
_replication_factor(config::shard_local_cfg().internal_topic_replication_factor.bind())
...
_replication_factor.watch([this] {
  ...
  ssx::spawn_with_gate(_gate, [this] { return update_topic() });
});
```